### DISCUSSION TITLE
Failed Jobs  Spatie\MediaLibrary\Conversions\Jobs\PerformConversionsJob

### DISCUSSION THREAD

**Discussion by JmRy** [spatie/laravel-medialibrary#3625] [Q&A]
Hello all, 

I have had this error since updating to Media-Library v11 on a production machine.

```
Spatie\Image\Exceptions\CouldNotLoadImage: Could not load image at path `/var/www/vhosts/xxxxxx.fr/www/storage/media-library/temp/z7S2l91B8lujv2HIyYJwQnX7eMEwOkzV/CcOfGs1bbXFG1yUp8yvZH4ROSkuyKZvVthumb.webp` in /var/www/vhosts/xxxxx.fr/www/vendor/spatie/image/src/Exceptions/CouldNotLoadImage.php:11
```

I'm going around in circles and I don't understand why I have this error. 

On my dev machine under Docker, this code works perfectly! Moreover, the same code works without problems in production on another server/site.

- It is not a permissions problem. The file exists in medialibrary/temp directory, but it is created empty!
- It is not a memory problem. With memory_limit = -1, it also bugs!
- It is not a queue problem. With ->noQueue or QUEUE_CONVERSIONS_BY_DEFAULT=false, it also bugs!
- It is not an issue with the original image. Regardless of its size, weight, or format,  it also bugs!

In short, I need help. 

Thank you to the first person who understands or has a lead. 😀🙏

```
PHP8.3
Laravel 10.48
spatie/browsershot 4.0.4
spatie/image  3.6.3 
spatie/laravel-medialibrary   11.5.2
```

My conversion is called by a job

```
public function handle(): void
{
    $savepath = storage_path('captures/cp'.$this->camping->id.'-w.webp');
    try {
        $this->camping->addMedia($savepath)->toMediaCollection('web');
        $this->camping->save();
    } catch (\Exception|FileDoesNotExist|FileIsTooBig $e) {
        FailedCaptures::create(['camping_id' => $this->camping->id, 'url' => $this->camping->web_url, 'exception' => sprintf('WebStore fail : Exception  %s %s', $e->getCode(), $e->getMessage())]);
        $this->fail();
    }
}
```

The variable $savepath contains the captured file which exists in the directory and it is indeed an image of several bytes.

The conversion is declared in the Camping model

```
public function registerMediaCollections(): void
{
  $this->addMediaCollection('web')
      ->singleFile()
      ->useFallbackUrl('/i/np270.png')
      ->useFallbackPath(public_path('/i/np270.png'));
}

public function registerMediaConversions(?Media $media = null): void
{
  $this->addMediaConversion('thumb')
      ->fit( Fit::Crop,530, 315)
      ->format('webp')
      ->withResponsiveImages()
      ->optimize()
      ->performOnCollections('web');
}
```

**Comment by dircm** [comment]
I have a slightly similar issue with my staging and production servers (Ubuntu 22.04.4 LTS).  My local dev env is Macbook.

My conversion code was generating a `Cannot assign __PHP_Incomplete_Class to property Spatie\MediaLibrary\Conversions\Conversion::$manipulations` exception whenever `Spatie\\MediaLibrary\\Conversions\\Jobs\\PerformConversionsJob` ran.

My conversion code :

```php
$this->addMediaCollection('images')
    ->useFallbackUrl(asset('images/Aircraft.jpeg'))
    ->useFallbackPath(public_path('images/Aircraft.jpeg'))
    ->withResponsiveImages()
    ->registerMediaConversions(function (?Media $media = null) {
        $this
            ->addMediaConversion('thumb')
            ->width(80);
        $this
            ->addMediaConversion('pdf')
            ->width(200);
    });
```
Disabling the optimizers while I figure this out has prevented the job from failing for now..

```php
$this->addMediaCollection('images')
    ->useFallbackUrl(asset('images/Aircraft.jpeg'))
    ->useFallbackPath(public_path('images/Aircraft.jpeg'))
    ->withResponsiveImages()
    ->registerMediaConversions(function (?Media $media = null) {
        $this
            ->addMediaConversion('thumb')
            ->width(80)
            ->nonOptimized();
        $this
            ->addMediaConversion('pdf')
            ->width(200)
            ->nonOptimized();
    });

```
